src/core/qgsapplication.cpp

```diff
@@ -2473,457 +2473,411 @@ QgsDataItemProviderRegistry *QgsApplication::dataItemProviderRegistry()
     mAnnotationItemRegistry->populate();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup labeling engine rule registry" ) );
-    mLabelingEngineRuleRegistry = new QgsLabelingEngineRuleRegistry();
+    mLabelingEngineRuleRegistry = std::make_unique<QgsLabelingEngineRuleRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup sensor registry" ) );
-    mSensorRegistry = new QgsSensorRegistry();
+    mSensorRegistry = std::make_unique<QgsSensorRegistry>();
     mSensorRegistry->populate();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup 3D symbol registry" ) );
-    m3DSymbolRegistry = new Qgs3DSymbolRegistry();
+    m3DSymbolRegistry = std::make_unique<Qgs3DSymbolRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup 3D renderer registry" ) );
-    m3DRendererRegistry = new Qgs3DRendererRegistry();
+    m3DRendererRegistry = std::make_unique<Qgs3DRendererRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup external storage registry" ) );
-    mExternalStorageRegistry = new QgsExternalStorageRegistry();
+    mExternalStorageRegistry = std::make_unique<QgsExternalStorageRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup profile source registry" ) );
-    mProfileSourceRegistry = new QgsProfileSourceRegistry();
+    mProfileSourceRegistry = std::make_unique<QgsProfileSourceRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup network content cache" ) );
-    mNetworkContentFetcherRegistry = new QgsNetworkContentFetcherRegistry();
+    mNetworkContentFetcherRegistry = std::make_unique<QgsNetworkContentFetcherRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup layout check registry" ) );
-    mValidityCheckRegistry = new QgsValidityCheckRegistry();
+    mValidityCheckRegistry = std::make_unique<QgsValidityCheckRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup classification registry" ) );
-    mClassificationMethodRegistry = new QgsClassificationMethodRegistry();
+    mClassificationMethodRegistry = std::make_unique<QgsClassificationMethodRegistry>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup bookmark manager" ) );
-    mBookmarkManager = new QgsBookmarkManager( nullptr );
+    mBookmarkManager = std::make_unique<QgsBookmarkManager>( nullptr );
     profiler->end();
   }
   {
     profiler->start( tr( "Setup tile download manager" ) );
-    mTileDownloadManager = new QgsTileDownloadManager();
+    mTileDownloadManager = std::make_unique<QgsTileDownloadManager>();
     profiler->end();
   }
   {
     profiler->start( tr( "Setup scalebar registry" ) );
-    mScaleBarRendererRegistry = new QgsScaleBarRendererRegistry();
+    mScaleBarRendererRegistry = std::make_unique<QgsScaleBarRendererRegistry>();
     profiler->end();
   }
 }
 
 QgsApplication::ApplicationMembers::~ApplicationMembers()
 {
-  delete mStyleModel;
```

Comment:
Destruction order here is very important -- can you add explicit .reset calls here in order to ensure we don't get crashes on exit?

Reply:
Done